firebase-sessions/src/test/kotlin/com/google/firebase/sessions/api/FirebaseSessionsDependenciesTest.kt

```diff
@@ -0,0 +1,127 @@
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withTimeout
+import org.junit.After
+import org.junit.Assert.assertThrows
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+class FirebaseSessionsDependenciesTest {
+  @After
+  fun cleanUp() {
+    // Reset any registered subscribers after each test.
+    FirebaseSessionsDependencies.reset()
+  }
+
+  @Test
+  fun register_dependencyAdded_canGet() {
+    FirebaseSessionsDependencies.register(crashlyticsSubscriber)
+
+    assertThat(FirebaseSessionsDependencies[CRASHLYTICS]).isEqualTo(crashlyticsSubscriber)
+  }
+
+  @Test
+  fun register_alreadyRegistered_throws() {
+    // Register the first time, no problem.
+    FirebaseSessionsDependencies.register(crashlyticsSubscriber)
+
+    val thrown =
+      assertThrows(IllegalArgumentException::class.java) {
+        // Attempt to register the same subscriber a second time.
+        FirebaseSessionsDependencies.register(crashlyticsSubscriber)
+      }
+
+    assertThat(thrown).hasMessageThat().contains("Subscriber CRASHLYTICS already registered")
+  }
+
+  @Test
+  fun getSubscriber_dependencyAdded_notRegistered_throws() {
+    val thrown =
+      assertThrows(IllegalStateException::class.java) { FirebaseSessionsDependencies[CRASHLYTICS] }
+
+    assertThat(thrown).hasMessageThat().contains("Subscriber CRASHLYTICS has not been registered")
+  }
+
+  @Test
+  fun getSubscriber_notDepended_throws() {
+    val thrown =
+      assertThrows(IllegalStateException::class.java) {
+        // Performance was never added as a dependency.
+        FirebaseSessionsDependencies[PERFORMANCE]
+      }
+
+    assertThat(thrown).hasMessageThat().contains("Cannot get dependency PERFORMANCE")
+  }
+
+  @Test
+  fun addDependencyTwice_throws() {
+    val thrown =
+      assertThrows(IllegalArgumentException::class.java) {
+        // CRASHLYTICS has already been added. Attempt to add it again.
+        FirebaseSessionsDependencies.addDependency(CRASHLYTICS)
+      }
+
+    assertThat(thrown).hasMessageThat().contains("Dependency CRASHLYTICS already added")
+  }
+
+  @Test
+  fun getSubscribers_waitsForRegister(): Unit = runBlocking {
+    // Wait a few seconds and then register.
+    launch {
+      delay(2.seconds)
+      FirebaseSessionsDependencies.register(crashlyticsSubscriber)
+    }
+
+    // Block until the register happens.
+    val subscribers = runBlocking { FirebaseSessionsDependencies.getSubscribers() }
+
+    assertThat(subscribers).containsExactly(CRASHLYTICS, crashlyticsSubscriber)
+  }
+
+  @Test(expected = TimeoutCancellationException::class)
+  fun getSubscribers_neverRegister_waitsForever() = runTest {
+    // The register never happens, wait until the timeout.
+    withTimeout(2.seconds) { FirebaseSessionsDependencies.getSubscribers() }
+  }
+
+  companion object {
+    init {
+      // Add only Crashlytics as a dependency, not Performance.
```

Comment:
Why not Performance?

Reply:
This is no longer needed now the new `reset` method clears depedencies, it is easier to create more isolated tests.